cmd/insights/contributors.go

```diff
@@ -0,0 +1,289 @@
+			if err != nil {
+				errorChan <- err
+				return
+			}
+			if repoContributorInsights == nil {
+				return
+			}
+			repoContributorInsights.RenderTable()
+		}(repoURL)
+	}
+	go func() {
+		waitGroup.Wait()
+		close(doneChan)
+	}()
+	select {
+	case <-doneChan:
+		return nil
+	case err = <-errorChan:
+		return err
+	}
+}
+
+type repositoryContributorsInsights struct {
+	RepoID  int32
+	RepoURL string
+	New     []string
+	Recent  []string
+	Churn   []string
+	Repeat  []string
+}
+
+func (rci *repositoryContributorsInsights) RenderTable() {
+	rows := []table.Row{
+		{"New contributors", strconv.Itoa(len(rci.New))},
+		{"Recent contributors", strconv.Itoa(len(rci.Recent))},
+		{"Churn contributors", strconv.Itoa(len(rci.Churn))},
+		{"Repeat contributors", strconv.Itoa(len(rci.Repeat))},
+	}
+	columns := []table.Column{
+		{
+			Title: "Repository URL",
+			Width: 20,
+		},
+		{
+			Title: rci.RepoURL,
+			Width: len(rci.RepoURL),
+		},
+	}
+	styles := table.DefaultStyles()
+	styles.Header.MarginTop(1)
+	styles.Selected = lipgloss.NewStyle()
+	repoTable := table.New(
+		table.WithColumns(columns),
+		table.WithRows(rows),
+		table.WithHeight(len(rows)),
+		table.WithStyles(styles),
+	)
+	fmt.Println(repoTable.View())
+}
+
+type ErrRepoNotFoundOrIndexed struct {
+	RepoURL string
+}
+
+func (e *ErrRepoNotFoundOrIndexed) Error() string {
+	return fmt.Sprintf("repository %s is either non-existent or has not been indexed yet", e.RepoURL)
+}
+
+func NewErrRepoNotFoundOrIndexed(repoURL string) *ErrRepoNotFoundOrIndexed {
+	return &ErrRepoNotFoundOrIndexed{
+		RepoURL: repoURL,
+	}
+}
+
+func findRepositoryByOwnerAndRepoName(ctx context.Context, apiClient *client.APIClient, repoURL string) (*client.DbRepo, error) {
+	owner, repoName, err := repoquery.GetOwnerAndRepoFromURL(repoURL)
+	if err != nil {
+		return nil, fmt.Errorf("could not extract owner and repo from url: %w", err)
+	}
+	repo, response, err := apiClient.RepositoryServiceAPI.FindOneByOwnerAndRepo(ctx, owner, repoName).Execute()
+	if err != nil {
+		if response.StatusCode == http.StatusNotFound {
+			return nil, NewErrRepoNotFoundOrIndexed(repoURL)
+		}
+		return nil, err
+	}
+	return repo, nil
+}
+
+func findAllRepositoryContributors(ctx context.Context, opts *Options, repoURL string) (*repositoryContributorsInsights, error) {
+	repo, err := findRepositoryByOwnerAndRepoName(ctx, opts.APIClient, repoURL)
+	if err != nil {
+		var errRepoNotFoundOrIndexed *ErrRepoNotFoundOrIndexed
+		if errors.As(err, &errRepoNotFoundOrIndexed) {
+			fmt.Println("ignoring repository issue:", errRepoNotFoundOrIndexed)
+			return nil, nil
+		}
+		return nil, fmt.Errorf("could not get contributor insights for repository %s: %w", repoURL, err)
+	}
+	repoContributorInsights := &repositoryContributorsInsights{
+		RepoID:  repo.Id,
+		RepoURL: repoURL,
+	}
+	var (
+		errorChan = make(chan error)
+		doneChan  = make(chan bool)
+		waitGroup = new(sync.WaitGroup)
+	)
+	waitGroup.Add(4)
```

Comment:
I think I'd prefer if we explicitly added `1` each time for each `go func` block. Otherwies, if another gets added, it won't be clear to other contributors what's going on here.